blockly/blocks/arduino/io.js

```diff
@@ -191,3 +191,60 @@ Blockly.Blocks['io_highlow'] = {
     return Blockly.Types.BOOLEAN;
   }
 };
+
+Blockly.Blocks['io_tone'] = {
+  init: function() {
+    this.appendDummyInput()
+        .appendField("Set tone on pin #")
+        .appendField(new Blockly.FieldDropdown(
+            Blockly.Arduino.Boards.selected.digitalPins), "TONEPIN");
+    this.appendValueInput("FREQUENCY")
+        .setCheck("Number")
+        .appendField("at frequency (in range 31 - 65535)");
```

Comment:
This text might make the block quite long, could we move the range into the tooltip? Then we should do a check in the code generator to ensure the range is correct or add a block warning.
(Don't worry to much about that, I can always edit it afterwards).